cdap-api/src/main/java/co/cask/cdap/api/app/ApplicationContext.java

```diff
@@ -30,7 +32,8 @@
   /**
    * Get the configuration object.
    *
-   * @return configuration object
+   * @return application configuration provided during application creation, else null
```

Comment:
we should be able to return an empty T I think.  With a future PR that will fail app creation if required fields are not present, similar to what happens with PluginConfig today.